uwblueprint/sistema
# Jason/sidebar filter

# Notion ticket link
https://www.notion.so/uwblueprintexecs/Class-type-location-filter-4714aeb1cf8944dfae4ce090669daf1f?pvs=4

[Class Type / Location Filter](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description

- Used useEffect to fetch locations and subjects data from APIs on component mount, ensuring necessary data is available before rendering.
- Set all checkboxes to be selected by default by initializing selectedLocations and selectedSubjects with all IDs from fetched data.
- Each checkbox’s isChecked property updates based on whether its id is in the selectedLocations or selectedSubjects arrays.
- Passed onSubjectChange and onLocationChange callbacks to manage selection changes, keeping the parent component in sync with the selected filters.



## Steps to test

1. Initial Load Test:
	•	Verify that when the component loads, all checkboxes for both locations and subjects are checked by default.
	•	Confirm that the correct API endpoints are called and that data is correctly stored in the component state
2. Checkbox State Test:
	•	Uncheck and Recheck a location or subject checkbox and confirm that the checkbox is unchecked visually and the functionality works
3.	Callback Functionality Test:
	•	Confirm that the onSubjectChange and onLocationChange callbacks are called with the correct id and isSelected values whenever a checkbox state changes.
4.	Performance and Re-Render Test:
	•	Check that the component only fetches data on the initial load and doesn’t make additional API calls on re-renders.



## What should reviewers focus on?

-	Ensure that data fetching for locations and subjects is correctly implemented, and all checkboxes are checked by default
- Confirm that the onSubjectChange and onLocationChange callbacks are functioning correctly
- Check that error handling in data fetching is robust

## Checklist

- [ X] My PR name is descriptive and in imperative tense
- [ X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ X] I have run the appropriate linter(s)
- [ X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
